habitat-lab/habitat/sims/habitat_simulator/sim_utilities.py

```diff
@@ -139,6 +139,140 @@ def get_ao_global_bb(
     return cumulative_global_bb
 
 
+def get_bb_for_object_id(
+    sim: habitat_sim.Simulator,
+    obj_id: int,
+    ao_link_map: Dict[int, int] = None,
+    ao_aabbs: Dict[int, mn.Range3D] = None,
+) -> Tuple[mn.Range3D, mn.Matrix4]:
+    """
+    Wrapper to get a bb and global transform directly from an object id.
+    Handles RigidObject and ArticulatedLink ids.
+
+    :param sim: The Simulator instance.
+    :param obj_id: The integer id of the object or link.
+    :param ao_link_map: A pre-computed map from link object ids to their parent ArticulatedObject's object id.
+    :param ao_aabbs: A pre-computed map from ArticulatedObject object_ids to their local bounding boxes. If not provided, recomputed as necessary.
```

Comment:
for my understanding, this is bounding box for a closed articulated object?

Reply:
This is a map of all AO bounding boxes for the scene. It includes the state of the links when computed. 
Typically, if pre-computed at initialization it will be for closed AOs (unless non-default AO states are specified in the scene instance config).
If it is computed in this function or re-computed during simulation, it will include the open state of the receptacles.